pallets/parachain-staking/src/benchmarks.rs

```diff
@@ -913,7 +913,157 @@ benchmarks! {
+		// INITIALIZE DELEGATIONS
+		let mut col_del_count: BTreeMap<T::AccountId, u32> = BTreeMap::new();
+		collators.iter().for_each(|x| {
+			col_del_count.insert(x.clone(), 0u32);
+		});
+		let mut delegators: Vec<T::AccountId> = Vec::new();
+		let mut remaining_delegations = if total_delegations > max_delegators_per_collator {
+			for j in 1..(max_delegators_per_collator + 1) {
+				let seed = USER_SEED + j;
+				let delegator = create_funded_delegator::<T>(
+					"delegator",
+					seed,
+					min_candidate_stk::<T>() * 1_000_000u32.into(),
+					collators[0].clone(),
+					true,
+					delegators.len() as u32,
+				)?;
+				delegators.push(delegator);
+			}
+			total_delegations - max_delegators_per_collator
+		} else {
+			for j in 1..(total_delegations + 1) {
+				let seed = USER_SEED + j;
+				let delegator = create_funded_delegator::<T>(
+					"delegator",
+					seed,
+					min_candidate_stk::<T>() * 1_000_000u32.into(),
+					collators[0].clone(),
+					true,
+					delegators.len() as u32,
+				)?;
+				delegators.push(delegator);
+			}
+			0u32
+		};
+		col_del_count.insert(collators[0].clone(), delegators.len() as u32);
+		// FILL remaining delegations
+		if remaining_delegations > 0 {
+			for (col, n_count) in col_del_count.iter_mut() {
+				if n_count < &mut (delegators.len() as u32) {
+					// assumes delegators.len() <= MaxDelegatorsPerCandidate
+					let mut open_spots = delegators.len() as u32 - *n_count;
+					while open_spots > 0 && remaining_delegations > 0 {
+						let caller = delegators[open_spots as usize - 1usize].clone();
+						if let Ok(_) = Pallet::<T>::delegate(RawOrigin::Signed(
+							caller.clone()).into(),
+							col.clone(),
+							<<T as Config>::MinDelegatorStk as Get<BalanceOf<T>>>::get(),
+							*n_count,
+							collators.len() as u32, // overestimate
+						) {
+							*n_count += 1;
+							remaining_delegations -= 1;
+						}
+						open_spots -= 1;
+					}
+				}
+				if remaining_delegations == 0 {
+					break;
+				}
+			}
+		}
+		// STORE starting balances for all delegators
+		let delegator_starting_balances: Vec<(
+			T::AccountId,
+			<<T as Config>::Currency as Currency<T::AccountId>>::Balance
+		)> = delegators.iter().map(|x| (x.clone(), T::Currency::free_balance(&x))).collect();
+
+		// rather than roll through rounds in order to initialize the storage we want, we set it
+		// directly and then call pay_one_collator_reward directly.
+
+		let round_for_payout = 5;
+		<DelayedPayouts<T>>::insert(&round_for_payout, DelayedPayout {
+			round_issuance: 100u32.into(),
+			total_staking_reward: 100u32.into(),
+			collator_commission: Perbill::from_rational(5u32, 100u32),
+		});
+
+		let mut delegations: Vec<Bond<T::AccountId, BalanceOf<T>>> = Vec::new();
+		delegations.push(Bond {
+			owner: sole_collator.clone(),
+			amount: 100u32.into(),
+		});
+
+		<AtStake<T>>::insert(round_for_payout, &sole_collator, CollatorSnapshot {
+			bond: 0u32.into(),
+			delegations,
+			total: 0u32.into(),
+		});
```

Comment:
This benchmark is testing the payout for a collator with a single delegation of itself to itself.

It is NOT testing the payout for a collator full of delegations. This is why the weight function does not depend on the number of delegations complexity parameter `y`.

This should be fixed to at least fill the collators delegations manually. The previous benchmarking did all the delegations via the extrinsics and filled up the delegations for all collators selected.